WooCommerce/WooCommerceTests/Stripe Integration Tests/StripeCardReaderIntegrationTests.swift

```diff
@@ -0,0 +1,53 @@
+import XCTest
+import Combine
+@testable import WooCommerce
+@testable import Hardware
+
+
+/// Integration tests for the integration with the Stripe Terminal SDK.
+/// We want to reach into the ServiceLocator in all tests
+/// because these are integration test, and we do not want to mock anything,
+/// at this point, other than the actual hardware
+final class StripeCardReaderIntegrationTests: XCTestCase {
+    private var cancellables: Set<AnyCancellable>!
+
+    override func setUp() {
+        super.setUp()
+        cancellables = []
+    }
+
+    /// The observed behaviour is that when discovery starts, the Stripe SDK will
+    /// call into its delegate with an empty array of "discovered readers".
+    /// Later on, it will call into its delegate again everytime it discovers a reader
+    /// passing the full list of readers discovered
+    func test_start_discovery_updates_discovered_readers_at_least_twice() {
+        let discoveredReaders = expectation(description: "Discovered Readers publishes first, an empty array, and then the actual reader(s) discovered")
+
+        let readerService = ServiceLocator.cardReaderService
+
+        readerService.discoveredReaders.sink { completion in
+            discoveredReaders.fulfill()
+        } receiveValue: { readers in
+            // The Stripe Terminal SDK published an empty list of discovered readers first
+            // and it will continue publishing as new readers are discovered.
+            // So we ignore the first call to receiveValue, and perform the test on the first call to
+            // receive value that is receiving a non-empty array.
```

Comment:
nit: if we always want to drop the first call, can also use [`dropFirst(1)`](https://developer.apple.com/documentation/combine/fail/dropfirst(_:))?

Reply:
Good call!